compose/service.py

```diff
@@ -667,7 +667,10 @@ def _get_container_host_config(self, override_options, one_off=False):
 
         devices = options.get('devices', None)
 
+        version = os.environ.get('COMPOSE_API_VERSION', '1.19')
+
         return create_host_config(
+            version=version,
```

Comment:
Yep, I agree with you. I did think of doing that refactoring while I was in there and the reason I didn't do these was I'm already now 3 levels away from the original bug I was working on! So decided I had to draw a line somewhere and decided to stop.